peerconnection.go

```diff
@@ -1632,34 +1633,41 @@ func (pc *PeerConnection) undeclaredMediaProcessor() {
 			if atomic.AddUint64(&simulcastRoutineCount, 1) >= simulcastMaxProbeRoutines {
 				atomic.AddUint64(&simulcastRoutineCount, ^uint64(0))
 				pc.log.Warn(ErrSimulcastProbeOverflow.Error())
+				pc.dtlsTransport.storeSimulcastStream(stream)
 				continue
 			}
 
 			go func(rtpStream io.Reader, ssrc SSRC) {
-				pc.dtlsTransport.storeSimulcastStream(stream)
-
 				if err := pc.handleIncomingSSRC(rtpStream, ssrc); err != nil {
 					pc.log.Errorf(incomingUnhandledRTPSsrc, ssrc, err)
+					pc.dtlsTransport.storeSimulcastStream(stream)
```

Comment:
Only store stream if it is not handled, this change can shorten the stream's lifecycle and can be recycled earlier if the stream is closed by renegotiation, since stream has a reference to the buffer, so this change can decrease memory usage in a long run session.